managedtenants/data/metadata.schema.yaml

```diff
@@ -210,6 +210,33 @@ properties:
       secretNamespace:
         type: string
         pattern: ^[A-Za-z0-9][A-Za-z0-9-]{0,60}[A-Za-z0-9]$
+  deadmanssnitch:
+    type: object
+    additionalProperties: false
+    description: "Denotes the Deadmans Snitch Configuration which is supposed to be setup alongside the Addon."
+    required:
+    - tags
+    properties:
```

Comment:
I'm under the impression we are exposing way more than the minimum required. The only properties that seem relevant for the tenants to customize are `tags` and `targetSecretRef`. Everything elase seems to be templated out of the other addon.yaml properties...

Reply:
Even I felt so but when I looked at the managed-tenants gitlab repository, there were some (or rather many) add-ons whose `deadmansnitch.yaml.j2`   consisted of custom `clusterDeploymentSelector` , `snitchNamePostFix`, `targetSecretRef`, etc. (all fields except dmsApiKeyRef for some or other addon) which didn't comply well with their addon metadata parameters (addon.yaml) the way they were supposed to.

For examples of such add-ons, you can check out:
https://gitlab.cee.redhat.com/service/managed-tenants/-/merge_requests/1507

and take a look at the Addon metadata-s in the MR where I have specified more fields under `deadmanssnitch` field apart from `tags`.

That's why I made the `deadmanssnitch` section tweakable for now so that at least we can proceed to deprecate the `extraResources` and migrate the Addons (having dms configuration) to make use of `deadmanssnitch` field and get rid of `extraResources` field and all the `deadmansnitch.yaml.j2` files.
This gives us the freedom to later on incrementally fix all the addons' deadmanssnitch configuration in such a way that they don't end up relying on custom fields under `deadmanssnitch` section except `tags`.

And from the tenants' perspective, they need not worry about setting all these fields because all of the "tweakable" fields under `deadmanssnitch` section have rightful defaults set. So, they can just set `deadmanssnitch.tags` and rest of the dms configuration would be taken care of.

Let me know your thoughts :)

Reply:
looking at those, it makes sense what you did. We should probably cleanup those differences, I believe they have no reason to exist. But let's do one change at a time... we first do as you say, later we can look into talking to our tenants to drop all that custom configuration which is not really needed.